File: pinyougou-manager-web/src/main/java/com/pinyougou/manager/controller/BrandController.java
package com.pinyougou.manager.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.pojo.TbBrand;
import com.pinyougou.sellergoods.service.BrandService;
import entity.PageResult;
import entity.Result;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping("/brand")
public class BrandController {
    //调用远程服务使用reference关联注解
    @Reference
    private BrandService brandService;

    /**
     * 查询所有数据
     * @return
     */
    @RequestMapping("/findAll")
    public List<TbBrand> findAll() {
        return brandService.findAll();
    }

    /**
     *   分页查询返回全部列表
     * @param pageNum   --当前页码
     * @param pageSize  --每页显示数量
     * @return
     */
    @RequestMapping("/findByPage")
    public PageResult<TbBrand> findByPage(int pageNum,int pageSize) {
        return brandService.findByPage(pageNum, pageSize);
    }

    /**
     * 根据id查询品牌信息
     * @param id
     * @return
     */
    @RequestMapping("/findOne")
    public TbBrand findOne(Long id) {
        return brandService.findOne(id);
    }

    //***************************************除查询外，增加，删除，修改都需要捕捉异常并提示操作结果,返回值为Result********************
    /**
     * 增加品牌
     * @param brand     -- 品牌
     * @return
     */
    @RequestMapping("/addBrand")
    public Result addBrand(@RequestBody TbBrand brand) {
        try {
            brandService.addBrand(brand);
            return new Result(true, "增加成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "增加失败");
        }
    }

    /**
     * 更新品牌信息
     * @param brand
     * @return
     */
    @RequestMapping("/update")
    public Result upadate(@RequestBody TbBrand brand) {
        try {
            brandService.updateById(brand);
            return new Result(true, "修改成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"修改失败");
        }
    }

    /**
     * 根据id批量删除品牌信息
     * @param ids
     * @return
     */
    @RequestMapping("/delete")
    public Result delete(Long[] ids) {
        try {
            brandService.deleteByIds(ids);
            return new Result(true, "删除成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"删除失败");
        }
    }

    /**
     *   分页查询返回全部列表
     * @param pageNum   --当前页码
     * @param pageSize  --每页显示数量
     * @return
     */
    @RequestMapping("/search")
    public PageResult search(@RequestBody TbBrand brand,int pageNum,int pageSize) {
        return brandService.findByPage(brand,pageNum, pageSize);
    }



}

File: pinyougou-manager-web/src/main/webapp/js/controller/brandController.js
//品牌控制层
app.controller('brandController' ,function($scope,$controller,brandService){
    $controller('baseController',{$scope:$scope});//继承baseController.js,伪继承

    //读取列表数据绑定到表单中
    $scope.findAll=function(){
        brandService.findAll().success(
            function(response){
                $scope.list=response;
            }
        );
    }

    //分页
    $scope.findByPage = function (pageNum, pageSize) {
        brandService.findByPage(pageNum, pageSize).success(
            function (response) {
                $scope.list = response.rows;//获得当前页数据,对应PageResult中的rows
                $scope.paginationConf.totalItems = response.total;//更新总记录数,对应PageResult中的total
            }
        );
    }

    //查询品牌实体
    $scope.findOne=function (id) {
       brandService.findOne(id).success(
            function (response) {
                $scope.entity=response;
            }
        );
    }

    //增加品牌数据,*****注：因为与修改是同一编辑窗口，所以先进行方法判断，根据点击打开时是否附带id
    //保存和修改公用一个方法，所以service层的调用之前先进行方法的判断
    $scope.save = function () {
        var serviceMethod;//声明一个service层的的调用
        if ($scope.entity.id != null) {
            serviceMethod = brandService.update($scope.entity);
        } else {
            serviceMethod = brandService.addBrand($scope.entity);
        }
        serviceMethod.success(
            function (response) {
                if (response.success) {
                    alert(response.message);
                    $scope.reloadList();//重新加载
                } else {
                    alert(response.message);
                }
            }
        );
    }

    //获得需要批量删除的ids数组
    $scope.selectIds=[];
    //批量删除操作
    $scope.dele=function () {
        brandService.dele($scope.selectIds).success(
            function (response) {
                if (response.success) {
                    alert(response.message);
                    $scope.reloadList();//重新加载
                    $scope.selectIds=[];
                } else {
                    alert(response.message);
                }
            })
    }

    //条件查询
    $scope.searchEntity={};
    $scope.search=function (pageNum, pageSize) {
        brandService.search(pageNum, pageSize,$scope.searchEntity).success(
                function (response) {
                $scope.list = response.rows;//获得当前页数据,对应PageResult中的rows
                $scope.paginationConf.totalItems = response.total;//更新总记录数,对应PageResult中的total
            }
        );
    }



});
